File: introduction/ITP1_2_D-Circle in a Rectangle.js
process.stdin.resume();
process.stdin.setEncoding('utf8');
process.stdin.on('data', function (chunk) {
  var line = chunk.toString().split(" ");

  var w = parseInt(line[0]);
  var h = parseInt(line[1]);
  var x = parseInt(line[2]);
  var y = parseInt(line[3]);
  var r = parseInt(line[4]);

  if (r <= x && x <= w - r && r <= y && y <= h - r) {
    console.log('Yes');
  } else {
    console.log('No');
  }
});

File: introduction/ITP1_2_D-Circle in a Rectangle.cpp
#include <iostream>
using namespace std;

int main(void) {
  int w, h, x, y, r;
  cin >> w >> h >> x >> y >> r;
  if (r <= x && x <= w - r && r <= y && y <= h - r) {
    cout << "Yes" << endl;
  } else {
    cout << "No" << endl;
  }
  return 0;
}
